File: dotnet/Solid/RefactoredSolid/LSP/Example2/Domain/WorldPayPayment.cs
using RefactoredSolid.LSP.Example2.External;

namespace RefactoredSolid.LSP.Example2.Domain
{
    public class WorldPayPayment : Payment
    {
        private readonly string _username;
        private readonly string _password;
        private readonly string _productId;

        public WorldPayPayment(string username, string password, string productId)
        {
            _username = username;
            _password = password;
            _productId = productId;
        }

        public override bool Refund(decimal amount, string transactionId)
        {
            WorldPayWebService worldPayWebService = new WorldPayWebService();
            string response = worldPayWebService.MakeRefund(amount, transactionId, _username, _password, _productId);
            return response.Contains("Success");
        }
    }
}

File: java/liskov-plus-refactor/src/main/java/no/bouvet/solid/lsp/domain/WorldPayPayment.java
package no.bouvet.solid.lsp.domain;

import java.math.BigDecimal;

import no.bouvet.solid.lsp.external.WorldPayWebService;

public class WorldPayPayment extends Payment {

    private String username;
    private String password;
    private String productId;

    public WorldPayPayment(String username, String password, String productId) {
        this.username = username;
        this.password = password;
        this.productId = productId;
    }

    @Override
    public Boolean refund(BigDecimal amount, String transactionId) {
        WorldPayWebService worldPayWebService = new WorldPayWebService();
        String response = worldPayWebService.makeRefund(amount, transactionId, username, password, productId);
        return response.contains("Success");
    }

}
